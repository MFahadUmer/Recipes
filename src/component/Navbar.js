import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import logo from '../assets/logo.png';
import CategoriesContainer from '../containers/CategoriesContainer';

const Navbar = ({ eventHandler }) => (
  <div className="navbarContainer">
    <div className="navbar">
      <div className="logo">
        <Link to="/"><img className="logo" src={logo} alt="logo" /></Link>
      </div>
      <div className="nytCooking">
        NYT Cooking
      </div>
      <div>
        <CategoriesContainer eventHandler={eventHandler} />
      </div>
    </div>
  </div>
);

Navbar.propTypes = {
  eventHandler: PropTypes.func.isRequired,
};

export default Navbar;
