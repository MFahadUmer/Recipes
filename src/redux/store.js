import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import foodReducer from './food/foodReducer';

const store = createStore(foodReducer, composeWithDevTools(applyMiddleware(thunk)));

export default store;
