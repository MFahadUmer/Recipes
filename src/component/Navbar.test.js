import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import Navbar from './Navbar';
import '@testing-library/jest-dom';
import store from '../redux/store';

describe('Navbar Component Tests', () => {
  const eventHandler = jest.fn();

  it('should renders the text NYT Cooking', () => {
    render(
      <Provider store={store}>
        <Router>
          <Navbar eventHandler={eventHandler} />
        </Router>
      </Provider>,
    );
    expect(screen.getByText('NYT Cooking')).toBeTruthy();
  });

  it('matches the Navbar component snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Navbar eventHandler={eventHandler} />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
