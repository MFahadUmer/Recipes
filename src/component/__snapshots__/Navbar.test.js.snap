// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Tests matches the Navbar component snapshot 1`] = `
<div
  className="navbarContainer"
>
  <div
    className="navbar"
  >
    <div
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="logo"
          src="logo.png"
        />
      </a>
    </div>
    <div
      className="nytCooking"
    >
      NYT Cooking
    </div>
    <div>
      <select
        className="selectCategory"
        id="categories"
        name="categories"
        onChange={[MockFunction]}
      >
        <option
          className="selectCategory"
          value="All"
        >
          All
        </option>
      </select>
    </div>
  </div>
</div>
`;
